File: insertionSort.py
arr=[]
for i in range(10):
  inserimento=int(input('inserisci un numero '))
  arr.append(inserimento)
print(arr)

#function insertionSort
def insertionSort(arr):
  appo = 0
  i = 0
  for i in range(1,len(arr)):
    appo=arr[i]
    j = i-1
    while(j>=0) and (arr[j]> appo):
      arr[j+1] = arr[j]
      j = j -1
    arr[j+1] = appo
  return arr

#function call
print(insertionSort(arr))

File: insertionSort.c
#include <stdbool.h>
#include <stdio.h>
int main(){
    int arr[]={4,365,3,7,2,6,3,47};
    int dim = 8;
    insertion_Sort(arr,dim);
    return 0;

}


void insertion_Sort(int arr[], int size){
    int appo = 0;
    for(int i = 1;i<8; i++){
        appo = arr[i];
        int j = i - 1;
        while(j >= 0 && arr[j] > appo){
            arr[j+1] = arr[j];
            j = j -1;
        arr[j+1] = appo; 
        }
               
    }
    printf("\n\n stampa lista ordinata");
    for(int i = 0; i < size;i++){
        printf("%d \n",arr[i]);
    }    

}

